### PR TITLE
Avatar

### PR DESCRIPTION
**Pull request by MacMeDan**
Avatar class implementation. Added the ability to show initials on an avatar. Defaults to values for everything. Added Tests and a `Playground` so others can play with it.

### CODE REVIEW

**Comment by SD10** [comment]
@MacMeDan Thanks! I still have to review this but I only wish you could've submitted this to `v0.2.0` instead of `master` 😭 Any way we can do this? It may be better for me to review it first.

File: Sources/Avatar.swift [hunk 1]
```diff
@@ -24,28 +24,93 @@
...
+    
+    open func setBackground(color: UIColor) {
+        backgroundColor = color
+    }
+    
+    open func getImage() -> UIImage {
+        return imageView.image ?? UIImage()
+    }
+    
+    open func roundCorners(by: CGFloat?) {
```
Comment (SD10) [inline Sources/Avatar.swift:108]: I think adding an internal parameter `radius` would be great 😃

File: Sources/Avatar.swift [hunk 2]
```diff
@@ -24,28 +24,93 @@
...
+        initalsLabel.font = UIFont.systemFont(ofSize: size)
+        initalsLabel.textColor = color
+    }
+    
+    open func setBackground(color: UIColor) {
+        backgroundColor = color
+    }
+    
+    open func getImage() -> UIImage {
+        return imageView.image ?? UIImage()
```
Comment (SD10) [inline Sources/Avatar.swift:105]: Any specific reason why this doesn't return an optional?

File: Tests/AvatarTests.swift [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    
+    override func tearDown() {
+        super.tearDown()
+    }
+    
+    func testNoParams() {
+        let avatar = Avatar()
+        XCTAssertEqual(avatar.initals, "?")
+        XCTAssertEqual(avatar.initalsLabel.text, "?")
+        XCTAssertEqual(avatar.layer.cornerRadius, 0.0)
```
Comment (SD10) [inline Tests/AvatarTests.swift:26]: Looks like this is failing the tests on iOS 9+. 
Avatar has a default size of `30 x 30` resulting in a `cornerRadius` of `15.0`

File: .swiftlint.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
-line_length:
-  warning: 150
-  ignores_comments: true
 
 disabled_rules:
-  identifier_name
+- identifier_name
+- trailing_whitespace
+- line_length
```
Comment (SD10) [inline .swiftlint.yml:5]: I do like the `trailing_whitespace` rule 🤔 
Same with `line_length` but this is so trivial, let's leave it for now.

**Review by SD10 — CHANGES_REQUESTED** [review]
Great work, thank you for writing unit tests 😅
If you could please review the comments and fix the failing unit tests ❤️

File: Sources/Avatar.swift [hunk 5]
```diff
@@ -24,28 +24,93 @@
...
-
-    public let highlightedImage: UIImage?
-
-    public let placeholderImage: UIImage
-
-    public init(image: UIImage? = nil, highlightedImage: UIImage? = nil, placeholderImage: UIImage) {
-        self.image = image
-        self.highlightedImage = highlightedImage
-        self.placeholderImage = placeholderImage
+open class Avatar: UIView {
```
Comment (SD10) [inline Sources/Avatar.swift:27]: Do you think `AvatarView` would be a better name? Allowing people to identify its a `UIView` subclass?

**Comment by SD10** [comment]
I'm just going to push this off to `v0.3.0`. Sorry for the nightmare of opening an closing PR's.

File: Sources/Avatar.swift (hunk 5 above)
Comment (MacMeDan) [reply Sources/Avatar.swift:27]: Ya that is what I had it called originally and then changed it for simplicity but I am not married to the idea I think I will go back to avatar view.

File: .swiftlint.yml (hunk 4 above)
Comment (MacMeDan) [reply .swiftlint.yml:5]: So if you enable `trailing_whitespace` it throws a warning for blank line between methods. I like to separate each method with one empty line to make it easier to read. That is why i disabled it. 

screen shot 2017-08-05 at 8 12 26 pm

Comment (SD10) [reply .swiftlint.yml:5]: You can still have empty lines but just no spaces on the empty line 🤔

Comment (fbartho) [reply .swiftlint.yml:5]: For what it's worth, there's an option that helps with this in Xcode Preferences->Text Editing->Editing -- make sure the bottom of the two checkboxes in the following screenshot is checked: https://cl.ly/0d3C2E2B3x0A

**Comment by SD10** [comment]
Closing in favor of #31

**Pull request closed without merging**